runtime/data/script/kernel/serial.lua

```diff
@@ -0,0 +1,31 @@
+local Handle = require "handle"
+local serpent = require "serpent"
+local inspect = require "inspect"
+
+local Serial = {}
+
+local function resolve_handles(data)
```

Comment:
This function causes stack overflow if `data` has recursive reference. Mod data may have these recursive structure.

```lua
local table = {}
table.itself = table
resolve_handles(table)
```